common/src/main/java/com/wynntils/gui/screens/ScreenWrapper.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.gui.screens;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.utils.McUtils;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.components.Widget;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.gui.narration.NarratableEntry;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.Style;
+import net.minecraft.network.chat.TextComponent;
+import net.minecraft.util.FormattedCharSequence;
+import net.minecraft.world.inventory.tooltip.TooltipComponent;
+import net.minecraft.world.item.ItemStack;
+
+public class ScreenWrapper extends Screen {
+    private final Screen delegate;
+
+    private ScreenWrapper(Component component, Screen delegate) {
+        super(component);
+
+        this.delegate = delegate;
+    }
+
+    public static Screen create(Screen instance) {
+        return new ScreenWrapper(instance.getTitle(), instance);
+    }
+
+    public Screen getDelegate() {
+        return delegate;
+    }
+
+    private void failure(String method, Throwable e) {
+        WynntilsMod.error("Failure in " + delegate.getClass().getSimpleName() + "." + method + "()", e);
+        McUtils.sendMessageToClient(new TextComponent("Wynntils: Failure in " + method + " in "
+                + delegate.getClass().getSimpleName() + ". Screen forcefully closed."));
+        McUtils.mc().setScreen(null);
+    }
+
+    @Override
+    public void init(Minecraft minecraft, int width, int height) {
+        try {
+            delegate.init(minecraft, width, height);
+        } catch (Throwable t) {
+            failure("init", t);
+        }
```

Comment:
Why do we only catch stuff in init and render? mouse/keyboard handlers, could crash just like these. 
IMO also try/catch: `keyPressed`, `keyReleased`, `charTyped`, `mouseMoved`, `mouseClicked`, `mouseReleased`, `mouseDragged` and `mouseScrolled`. I hope this is a complete list.

Reply:
And about this... Do we wanna try/catch this init, or the other one? This is the one that MC calls right? And the other one is overriden to inject renderables (1.19 renamed the later one to `readdRendererables` or smth like that)

Reply:
As I said:

>Future possible improvements:
>
>  * Wrap more delegated methods in crash protection

I tried making a regexp to match all methods and insert the "try/catch/call `failure` with method name" pattern everywhere, but gave up. And I didn't want to do all copy paste manually.

Like 80-90% of the code in all Screens are in init and render, so the likelihood of crashes is by far biggest there. But I agree in principle that we should protect the rest as well. 

But your list is not so long. If you are satisfied with that list, I can add those as well.

Reply:
I never used the other methods, and I made a few screens :). So I think these should do.

Reply:
We definitely want to catch this init. I had quite a bit of an adventure trying to figure out how to get this to work wrt to init. I started by just capturing the parameterless init that is overridden. Since that was protected, I could not call it. *sigh* So I opened up with an accesswidener. Then I could not build since all our overrides where protected and now the superclass where public and you are not allowed to make accessability narrower. *sigh* So I removed the accesswidener and added a method invoker instead using mixins. Now that part worked. But then screens started crashing. Since the top-level init method setup the field `minecraft`, this was done for the wrapper, and not the delegate. I could have copied that, but it seemed wrong, so I changed the override to the top-level init instead, but that means I had to remove the final from it.

But now things work.

I have checked all our screens. I have skimmed through Screens.java in vanilla and tried to verify that they don't do anything else weird that might cause problems. So I think this is fine.

Reply:
Okay. Sounds good.

Reply:
> But your list is not so long. If you are satisfied with that list, I can add those as well.

Trust me, I added enough screens to know where I can crash them :)

Reply:
I am just working on one. Hence this PR review was done on 3-4 separate occasions. As I was writing the new screen, I realized what could go wrong here :)

Reply:
Ok, with one nights sleep I was able to figure out how to do the regexp, so I fixed all delegated calls.